pkg/costmodel/allocation_helpers.go

```diff
@@ -2353,18 +2352,18 @@ func getUnmountedPodForNamespace(window opencost.Window, podMap map[podKey]*pod,
 
 func calculateStartAndEnd(result *prom.QueryResult, resolution time.Duration, window opencost.Window) (time.Time, time.Time) {
 	// Start and end for a range vector are pulled from the timestamps of the
-	// first and final values in the range. There is no "offsetting" required
-	// of the start or the end, as we used to do. If you query for a duration
-	// of time that is divisible by the given resolution, and set the end time
-	// to be precisely the end of the window, Prometheus should give all the
-	// relevant timestamps.
-	//
-	// E.g. avg(kube_pod_container_status_running{}) by (pod, namespace)[1h:1m]
-	// with time=01:00:00 will return, for a pod running the entire time,
-	// 61 timestamps where the first is 00:00:00 and the last is 01:00:00.
+	// first and final values in the range.
 	s := time.Unix(int64(result.Values[0].Timestamp), 0).UTC()
 	e := time.Unix(int64(result.Values[len(result.Values)-1].Timestamp), 0).UTC()
 
+	// As of Prometheus v3, we have had to reintroduce the "offsetting" of the
+	// start time.
+	//
+	// E.g. avg(node_total_hourly_cost{}) by (node, provider_id)[1h:5m] with
+	// time=01:00:00 will return, for a node running the entire time, 12
+	// timestamps where the first is 00:05:00 and the last is 01:00:00.
+	s = s.Add(-resolution)
```

Comment:
what if I am running v2 and my node legitimately started at 10 mins past the hour? will it get walked back to 5 min past the hour?

Reply:
Yes, unfortunately that's correct. If you have Prometheus v2 and your node legitimately starts at `time=01:15:00`, this function will return a start time of `time=01:10:00` (assuming default `ETL_RESOLUTION_SECONDS`).

This only occurs if a node started sometime within the query window. Which is less bad than the current bug where all asset/allocation start times are affected regardless of query window.

Reply:
If needed, we could refactor to use the [`/api/v1/status/buildinfo`](https://prometheus.io/docs/prometheus/latest/querying/api/#build-information) Prometheus API to determine the Prometheus version the user is running?

Reply:
@thomasvn im just putting some thought here but i total agree with Alex, you need a more robust way to identify prometheus version 

perhaps you could leverage the /version end point of prometheus as https://prom.kubecost.xyz/version

Then identify major version to be 3.X.X and only then add offset of 5 minutes. @ameijer thoughts?

Reply:
@ameijer @avrodrigues5 Requesting another review! I've included logic to query/save the Prometheus version. Then only apply the offset if we are on a Prometheus version 3 or greater.